enumerables.rb

```diff
@@ -1,37 +1,72 @@
 module Enumerables
+
   def my_each
-    count = 0
-    while count < self.size do
-      yield(self[count])
-      count += 1
+    return to_enum unless block_given?
+
+    for i in 0..self.length
+      yield(self[i])
     end
-    self
   end
 
   def my_each_with_index
-    
+    return to_enum unless block_given?
+    my_index = 0
+    self.my_each do |i|
+      yield i, my_index
+      my_index += 1
+    end
   end
 
   def my_select
     my_arr = []
-    count = 0
-    while count < self .length
-      my_arr << (self[count]) if yield(self[count]) == true
-      count += 1
-    end
+    self.my_each{ |x| my_arr << x if yield(x) }
     my_arr
   end
 
-  def my_all?
-  end
+    def my_all?
+      self.my_each do |index|
+        return true unless block_given?
+        true_false = yield(index)
+        return false unless true_false
+      end
+      true
+      to_enum
+    end
 
-  def my_any?
+  def my_any?(elem = 0)
+    unless block_given?
+      if elem.instance_of?(Class)
+        my_each { |x| return true if x.is_a? elem }
+      elsif elem.instance_of?(Regexp)
+        my_each { |x| return true if elem.match?(x.to_s) }
+      elsif [nil, false].include?(elem)
+        my_each { |x| return true if x == elem }
+      end
+      return false
+    end
+    my_each { |x| return true if yield(x) }
+    false
   end
 
```

Comment:
- [ ] enumerables #my_any? returns false if the block ever returns a value false or nil
      Failure/Error: let(:false_block) { proc { |num| num > HIGHEST_VALUE } }

- [ ] enumerables #my_any? returns false if the block ever returns a value false or nil::range
      Failure/Error: expect(range.my_any?(&false_block)).to eq(range.any?(&false_block))

- [ ] enumerables #my_any? when no block or argument is given returns true if at least one of the collection is not false or nil
      Failure/Error: expect(true_array.my_any?).to be true_array.any?
      
        expected true
             got false

- [ ] enumerables #my_any? when a pattern other than Regex or a Class is given returns true if any of the collection matches the pattern
      Failure/Error: expect(words.my_any?('cat')).to be words.any?('cat')
      
        expected true
             got false